### ISSUE TITLE
Cannot use Context in Librispeech model

### ISSUE THREAD

**Issue by mapledxf** [wenet-e2e/wenet#1529]
Hi there,
Thanks for the awesome project. Now I am facing some Context related issue here:

First, I registered a context word like "TURN" which is not in the units.txt coming from the librispeech model, but there is a space-prefixed word "▁TURN 4622" in the units.txt.

So when I run the code, the context word "TURN" has been split into 2 ids: 4619 2930 which are "TRU" and "N" respectively. And in the decoding phase, the decoder will return the "▁TURN" id which is 4622, and that will cause the Context cannot be matched.

Any idea how I can solve that?

**Comment by robin1001** [comment]
Yes, there is some bug for English if you use BPE as modeling unit. @pengzhendong do you have time to fix it?

**Comment by pengzhendong** [comment]
> Yes, there is some bug for English if you use BPE as modeling unit. @pengzhendong do you have time to fix it?

I will fix it.

**Comment by mapledxf** [comment]
@pengzhendong 
So instead of calling SplitUTF8StringToWords, maybe we should use BPE to decode the text into words for English inputs.

**Comment by robin1001** [comment]
I think we can just add '▁' at the begin of English word, then use the same code as before.

**Comment by robin1001** [comment]
Fixed, please see https://github.com/wenet-e2e/wenet/pull/1540.
